libraries/AP_RCProtocol/AP_RCProtocol_CRSF.cpp

```diff
@@ -468,8 +471,14 @@ void AP_RCProtocol_CRSF::process_link_stats_frame(const void* data)
     } else if (rssi_dbm > 120) {
         _link_status.rssi = 0;
     } else {
+#ifndef IOMCU_FW
+        AP_RSSI* ap_rssi = AP::rssi();
+        float scale = ap_rssi ? ap_rssi->get_rssi_scale() : 1;
         // this is an approximation recommended by Remo from TBS
+        _link_status.rssi = constrain_int16(roundf((1.0f - (rssi_dbm - 50.0f) / 70.0f) * 255.0f) * scale, 0, 255);
```

Comment:
We should probably try to share more code here.  So create a temporary holding `roundf((1.0f - (rssi_dbm - 50.0f) / 70.0f) * 255.0f)`, multiply it by `scale` if appropriate then constrain the result into _link_status.rssi`.

I think you need to check `constrain_int16` is the correct call here; I'm pretty sure it's the argument type not the result type that matters (so you want `constrain_float`) - and that might mean you need to cast into `int16_t` afterwards.